public/app/features/plugins/loader/constants.ts

```diff
@@ -1,5 +1,5 @@
 export const SHARED_DEPENDENCY_PREFIX = 'package';
 export const LOAD_PLUGIN_CSS_REGEX = /^plugins.+\.css$/i;
 export const JS_CONTENT_TYPE_REGEX = /^(text|application)\/(x-)?javascript(;|$)/;
-export const ENDS_WITH_FILE_EXTENSION_REGEX = /\/?\.[a-zA-Z]{2,}$/;
-export const IS_SYSTEM_MODULE_REGEX = /System\.register\(/;
+export const AMD_MODULE_REGEX =
```

Comment:
This is a big noodle. Could you please add test cases that cover what this regex is aiming to cover in the check where we use it?

Reply:
@jackw BTW. Feel free to merge this PR to address the bug and open and issues to address those tests in a follow up PR. That way we can deliver this fix quickly.

Reply:
@academo I've added some tests in https://github.com/grafana/grafana/pull/74471/commits/316c781d7a4613bf303f9a921bec3ebd2112d30e